### ISSUE TITLE
False positive for SC1003 (escape single quote)

### ISSUE THREAD

**Issue by atomGit** [koalaman/shellcheck#2639]
#### For bugs
- SC1003
- shellcheck 0.9.0 + online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
declare -A a=( ['\']='%5C' )
```

#### Here's what shellcheck currently says:

`^-- SC1003 (info): Want to escape a single quote? echo 'This is how it'\''s done'.`

**Comment by Potherca** [comment]
Same issue:

```sh
cat "${jsonFile}" | tr -d '\\' | jq
```
